File: java_db_service/src/main/java/ru/mipt/osint_db_service/model/webParsing/WebFoundAddress.java
package ru.mipt.osint_db_service.model.webParsing;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;
import ru.mipt.osint_db_service.model.FoundAddressDto;

@Getter
@Setter
@Entity
@Table(name = "web_found_address", schema = "osint_web")
public class WebFoundAddress implements FoundAddressDto {
    @Id
    @Column(name = "id", nullable = false)
    private String id;

    @Column(name = "search_type", nullable = false, length = Integer.MAX_VALUE)
    private String searchType;

    @Column(name = "crypto_name", nullable = false, length = Integer.MAX_VALUE)
    private String cryptoName;

    @Column(name = "pattern_name", nullable = false, length = Integer.MAX_VALUE)
    private String patternName;

    @Column(name = "address", nullable = false, length = Integer.MAX_VALUE)
    private String address;

    @Column(name = "context", length = Integer.MAX_VALUE)
    private String context;

    @Column(name = "source", length = Integer.MAX_VALUE)
    private String source;

    @Column(name = "found_time", length = Integer.MAX_VALUE)
    private String foundTime;

    @Column(name = "is_approved")
    private Boolean approved;
}

File: Osint_web_service/src/main/kotlin/ru/bitok/osint/osint_web_service/entity/WebFoundAddress.kt
package ru.bitok.osint.osint_web_service.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.hibernate.annotations.ColumnDefault
import ru.bitok.osint.osint_web_service.util.AddressParser
import java.time.Instant

@Entity
@Table(name = "web_found_address", schema = "osint_web_v2")
class WebFoundAddress {
    @Id
    @Column(name = "address", nullable = false, length = Integer.MAX_VALUE)
    var address: String = ""

    @Column(name = "search_type", length = Integer.MAX_VALUE)
    var searchType: String = "WEB"

    @Column(name = "crypto_name", nullable = false, length = Integer.MAX_VALUE)
    var cryptoName: String = AddressParser.CryptoName.UNIVERSAL.name

    @Column(name = "pattern_name", nullable = false, length = Integer.MAX_VALUE)
    var patternName: String = "NOT_STATED"

    @Column(name = "valid_address", nullable = false)
    var validAddress: Boolean = false

    @Column(name = "create_time")
    var createTime: Instant = Instant.now()
}